### ISSUE TITLE
[Feature] Recipe multiplier

### ISSUE THREAD

**Issue by Eothred** [Teifun2/nextcloud-cookbook-flutter#167]
I've been using MicMun's version for a long while, unaware the two versions exist. One of the major inventions of this app in my eyes was the addition of the multiplier, where you could change the number of servings and all numbers in the ingredients list are multiplied accordingly. This is during reading the recipe, not during editing, so the recipe is not permanently changed for a different number of servings.

My most common example is my pizza dough recipe, which by default is for 2 thin pizzas. If I want a little thicker one day I maybe put in 2.5 servings, or if I'm making 5 pizzas I put in 5. Even if this multiplication is often simple (say 2x or 0.5x all numbers), it's still super nice to not have to do these in your head while tired or concentrating on other things as well.

**Comment by Slomo17** [comment]
I would like this feature too, but i think the problem is, that you can put anything into to ingredients field and the app does not know what to do with e.g. "one onion" , so the ingredients have to follow some conventions.

**Comment by Eothred** [comment]
If you try the feature in the other app you will see it is very flexible. I think it just finds all numbers and use a multiplier (you can edit the number of servings in view mode). I would think that is "too naive" way to do it and it requires more formal syntax in the recipe to work, but it works really well.

In the case where the recipe has "one onion", the user could then just edit the recipe and replace with 1 to have the feature enabled.

**Comment by Teifun2** [comment]
@Eothred i agree that this is one of the main features that is missing so far.

However if done i would like to do it right, and for this the schema from schema.org first needs to change sadly.

For a more indepth discussion see: 
https://github.com/nextcloud/cookbook/issues/116
